plugins/arkworks/src/constraint.rs

```diff
@@ -261,6 +271,22 @@ where
     }
 }
 
+impl<F, const BITS: usize> BitDecomposition<BITS, R1CS<F>> for FpVar<F>
+where
+    F: PrimeField,
+{
+    #[inline]
+    fn to_bits_le(&self, compiler: &mut R1CS<F>) -> [Boolean<F>; BITS] {
+        let _ = compiler;
+        assert_eq!(
+            BITS,
+            F::Params::MODULUS_BITS as usize,
+            "BITS must be equal to MODULUS BITS"
+        );
+        ToBitsGadget::to_bits_le(self).expect("Bit decomposition is not allowed to fail.").try_into().expect("Obtaining an array of size BITS from a vector of length BITS is not allowed to fail.")
+    }
+}
```

Comment:
we can't use `F::Params::MODULUS_BITS` here for `BitDecomposition<X, R1CS<F>>`?

Reply:
Why is that?

Reply:
rust doesn't let me do the clean implementation 
impl<F> BitDecomposition<F::Params::MODULUS_BITS, R1CS<F>> for FpVar<F>